/*******************************************************************************
 * FileName     : version.proto
 * Author       : liuhui <hui.liu.e@gmail.com>
 * Last Modified: 2016-06-14 00:59:16
 * Description  : 协议版本号
 ******************************************************************************/

 import public "MessageID.proto";

// S -> C 检查协议版本号
message ReqCheckVersion {

    optional MessageID  msg_id = 1 [default = MSG_CHECK_VERSION];

    required int32      iVersionNumber = 2 [default = 1];
}

// 版本验证结果
enum NOTIFY_VERSION_CHECK_RESULT {
    NVCR_SUCCESS    = 0;    // 版本一致
    NVCR_FAILED     = 1;    // 版本不符
}

// C -> S 版本检查结果
message NotifyVersionCheck {

    optional MessageID  msg_id = 1 [default = MSG_CHECK_VERSION];
    required NOTIFY_VERSION_CHECK_RESULT result = 2 [default = NVCR_SUCCESS];
    // Next available id: 1
}
